services/QuillLMS/app/services/demo/session_data.rb

```diff
@@ -87,8 +84,13 @@ def generate_files
       ACTIVITY_USER_PAIRS.map do |id_pair|
         activity_id, user_id = id_pair
 
-        ActivitySession.unscoped.find_by(activity_id: activity_id, user_id: user_id, is_final_score: true)
-      end.compact
+        # we only want to find the first one if it's a diagnostic because there should be no replays
+        if Activity.diagnostic_activity_ids.include?(activity_id)
+          ActivitySession.unscoped.find_by(activity_id: activity_id, user_id: user_id, state: ActivitySession::STATE_FINISHED)
+        else
+          ActivitySession.unscoped.where(activity_id: activity_id, user_id: user_id, state: ActivitySession::STATE_FINISHED)
+        end
+      end.flatten.compact
```

Comment:
This is purely stylistic, but we can use an early return pattern here inside a loop block the same way we do in methods except we use `next` instead of `return`:
```
ACTIVITY_USER_PAIRS.map do |id_pair|
  activity_id, user_id = id_pair

  # we only want to find the first one if it's a diagnostic because there should be no replays
  next ActivitySession.unscoped.find_by(...) if Activity.diagnostic_activity_ids.include?(activity_id)

  ActivitySession.unscoped.where(...)
end.flatten.compact
```